woodwork/tests/test_utils.py

```diff
@@ -566,3 +569,452 @@ def test_parse_logical_type_errors():
+
+
+def test_concat_different_table_types(sample_df_pandas, sample_df_dask):
+    sample_df_pandas.ww.init()
+    sample_df_dask.ww.init()
+
+    error = re.escape("cannot concatenate object of type '<class 'dask.dataframe.core.DataFrame'>'; only Series and DataFrame objs are valid")
+    with pytest.raises(TypeError, match=error):
+        concat([sample_df_dask, sample_df_pandas], axis=0)
+
+
+def test_concat_cols_inner_join(sample_df):
+    df1 = sample_df.loc[[0, 1, 2], ['id', 'full_name', 'age']]
+    df1.ww.init(index='id')
+    df2 = sample_df.loc[[2, 3], ['email', 'signup_date']]
+    df2.ww.init(time_index='signup_date')
+
+    inner_combined = concat([df1, df2], join='inner', axis=1)
+    assert set(inner_combined.columns) == {'id', 'email', 'full_name', 'age', 'signup_date'}
+    assert len(inner_combined) == 1
+    assert inner_combined.ww.index == 'id'
+    assert inner_combined.ww.time_index == 'signup_date'
+
+
+def test_concat_rows_inner_join_lose_index(sample_df):
+    df1 = sample_df[['signup_date', 'email', 'age']]
+    df1.ww.init(time_index='signup_date')
+
+    df2 = sample_df[['id', 'email', 'signup_date', 'age']]
+    df2.ww.init(index='id', time_index='signup_date')
+
+    # If we lose a column in the inner join, we lose its typing information
+    inner_combined = concat([df1, df2], join='inner', axis=0)
+    assert set(inner_combined.columns) == {'email', 'age', 'signup_date'}
+    assert inner_combined.ww.index is None
+    assert inner_combined.ww.time_index == 'signup_date'
+
+
+def test_concat_rows_inner_join_lose_time_index(sample_df):
+    df1 = sample_df[['signup_date', 'email', 'age']]
+    df1.ww.init(time_index='signup_date')
+
+    df2 = sample_df[['id', 'email', 'age']]
+    df2.ww.init(index='id')
+
+    # If we lose a column in the inner join, we lose its typing information
+    inner_combined = concat([df1, df2], join='inner', axis=0)
+    assert set(inner_combined.columns) == {'email', 'age'}
+    assert inner_combined.ww.index is None
+    assert inner_combined.ww.time_index is None
+
+
+def test_concat_table_order(sample_df):
+    df1 = sample_df[['id', 'signup_date']]
+    df1.ww.init(index='id')
+    df2 = sample_df[['id', 'age']]
+    df2.ww.init(index='id')
+
+    combined_df1 = concat([df1.ww.copy(), df2.ww.copy()])
+    combined_df2 = concat([df2.ww.copy(), df1.ww.copy()])
+
+    assert set(combined_df1.columns) == set(combined_df2.columns)
+
+    assert list(combined_df1.columns) == ['id', 'signup_date', 'age']
+    assert list(combined_df2.columns) == ['id', 'age', 'signup_date']
+
+
+def test_concat_cols_all_series(sample_df):
+    age_series = ww.init_series(sample_df['age'], logical_type='Double')
+    combined_df = concat([sample_df['id'], age_series, sample_df['is_registered']], axis=1)
+
+    assert isinstance(combined_df.ww.schema, ww.table_schema.TableSchema)
+    assert set(combined_df.columns) == {'id', 'age', 'is_registered'}
```

Comment:
Think we should test the column order here like we did in the test immediately preceding this?

Reply:
I think `test_concat_table_order` is technically testing the relative column order for `axis=1` concatenation of multiple dataframes, and we can add column order when concatenating just series. 

But I think the equivalent of the original column order for row concatenation is to check concatenating columns where the indexes are the same but in different orders. I'll add a test for that